File: DIP/DIP_003/C++/DIP_003_decreasing_intensity_of_all_pixels.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;

int main() {
    
    cv::Mat img;
    
    // using the following argument we will load a preprocessed image
    // in this case RGB image will be converted into a gray scale image
    img = cv::imread("sophie_turner.jpg", IMREAD_GRAYSCALE) ;
    
    // changing some grayscale pixel values
    for (int k = 0; k<50; k++) {
        
        for(int j=0;j<img.rows;j++){

            for (int i=0;i<img.cols;i++){
            
                img.at<uchar>(j,i) = img.at<uchar>(j,i) * 0.97 ; //darker
            }

        }

        cv::imshow("Updated",img);
        cv::waitKey(40);
    }
    return 0;    
}

File: DIP/DIP_003/Python/DIP_003_decreasing_intensity_of_all_pixels.py
# import opencv library
import cv2
import numpy as np

# read the image
img = cv2.imread('003.jpg',0)

cv2.imshow("Grayscale_image_modified.jpg",img)
cv2.waitKey()
cv2.destroyAllWindows()

for k in range(0,50):
    img[:,:] = img[:,:] * 0.97
    # show the results
    cv2.imshow('Updated',img)
    cv2.waitKey(40)
    
# save the image
cv2.imwrite("Grayscale_image_modified.jpg",img)
cv2.destroyAllWindows()
# show the results
cv2.imshow('Updated',img)
cv2.destroyAllWindows()